tokio/src/io/util/copy.rs

```diff
@@ -241,10 +241,28 @@ cfg_io_util! {
     /// [`AsyncBufRead`]. You can use `copy_buf` with [`BufReader`] to change the
     /// buffer capacity.
     ///
+    /// # When to use async alternatives instead of `SyncIoBridge`
+    ///
+    /// If you are looking to use [`std::io::copy`] with a synchronous consumer
+    /// (like a `hasher` or compressor), consider using async alternatives instead of
+    /// wrapping the reader with [`SyncIoBridge`]. For example:
+    ///
+    /// - **Hashing**: Read data with [`read_to_end`] or in chunks with [`read`],
+    ///   then feed it to the `hasher` directly.
+    /// - **Compression**: Use an async compression library like [`async-compression`].
+    /// - **Parsing formats**: Read data into a `Vec<u8>` and use `from_slice`
+    ///   instead of `from_reader`.
+    ///
+    /// See the [`SyncIoBridge`] documentation for detailed examples and guidance.
```

Comment:
```suggestion
    /// wrapping the reader with [`SyncIoBridge`].
    /// See the [`SyncIoBridge`] documentation for detailed examples and guidance.
```

Reply:
I prefer to keep cross-links concise like what this PR did in `tokio/src/io/util/copy_buf.rs`, as downstream users can always click the link for more details.